webapp/packages/core-utils/src/getPathName.test.ts

```diff
@@ -0,0 +1,23 @@
+import { getPathName } from './getPathName';
+
+jest.mock('./getPathParts', () => ({
+  getPathParts: (path: string) => path.split('/'),
+}));
```

Comment:
not sure that we need to mock functions to isolate them, because in real world this is possibility that change in one function will break others but with mocks all other functions will pass tests

Reply:
unit test supposed to test only logic of current unit. if the test has some dependencies it is have to be mocked. this is the main point of unit tests

for example we have 2 functions:
`getPathParts`
`getPathName` which has dep of `getPathParts`

1st case when we covered everything with unit tests and mocked `getPathParts` in `getPathName` tests:
- if some of the test is fallen, we can actually say which one has a bug
- we test only the logic of current unit not thinking about what happens in dependency function

2nd case where we don't use mocks:
- if `getPathParts` has a bug. it is gonna affect other tests. instead of 1 fallen you can have **N** fallen tests. Imagine this function was used in **50 tests**. You need to spend some time to find the reason why the test is fell.
- it is not unit tests cause we also test interaction between modules, so it is more like some kind of integration testing

so what we have on a plate here:
- 1st approach is gonna show us where you for sure have no bug and where you have it (once it covered with tests). its will show us the bonuses once we have our code base mostly covered with unit tests
- 2nd approach is cool when we don't have a lot of coverage and we need to indicate that something somewhere in the fog went wrong. through the time it is gonna be a huge pain and tests will be useless, which will lead us to rewrite it to the 1st case way

so my call is to use mocks due to the fact it can save us time to find a place where something went wrong, and through the time will be more helpful with each test being written

I also suggest to have a discussion with the full frontend team what is our main approach for unit testing

Reply:
i see the point, let's leave it as is by now